test/Stats.test.js

```diff
@@ -257,7 +263,7 @@ describe("Stats", () => {
 			      ],
 			      "comparedForEmit": false,
 			      "emitted": true,
-			      "filteredRelated": undefined,
```

Comment:
Why it was changed?

Reply:
https://github.com/webpack/webpack/commit/13af6c25fc24dfd15e8c5a811dc45fcd11183273 refer to commit message...

I determined it should have been an array by looking at the condition a few lines after where it does `item.related || []`. However in the case where it never actually populates, the value remained as `undefined`. This is fine in theory but when you look at the stats output, the `undefined` got converted to an empty object. I'm not sure why that is the case though.

Another solution to this would be to fix how `related` is serialised but I'm not all that familiar with the webpack codebase (yet) to do that.

I don't believe it is correct to define `related` as an array or empty object as it would be confusing to anyone using the type definition as they wouldn't realise that empty object really means no related files, which is best represented by either not defining the value (which is the original intent, but then the serialiser is wrong) or using an empty array.

Reply:
related should be always array, otherwise it is breaking change

Reply:
Sorry if I didn't make it clear, that's exactly what I was saying.

Presently in master, when you look at the stats output for an asset with no related items, the result is something like this...

``` js
{
  // ...other asset props
  "related": {}
}
```

But what we really want is
``` js
{
  // ...other asset props
  "related": []
}
```

The mentioned commit fixes it so that is now the latter (second one).

Reply:
`filteredRelated` would be undefined if `item.related` stayed as `undefined`. However, this links back to the issue mentioned above that when `item.related` gets serialised, it goes from `undefined` to `{}`.

I think this is probably a separate issue. Do you want me to revert the commit, put the `{}` as part of the `related` type, and then raise a bug to say that `related` is being incorrectly serialized to `{}` whereas it should either be not defined or `[]`?

Reply:
Please do not touch logic, only types, otherwise it will not merged, I do mistake, it should be `{}`

Reply:
I've reverted the commit and left a comment on the `related` definition.